src/v/pandaproxy/schema_registry/seq_writer.cc

```diff
@@ -71,110 +71,299 @@ ss::future<> seq_writer::read_sync_inner() {
+        // originally read... remove magic field now that we aren't aiming
+        // for topic-level compatibility?
+        if (s.delete_subject) {
+            auto key = delete_subject_key{
+              .seq{s.seq}, .node{s.node}, .sub{sub}};
+            keys.push_back(key);
+            rb.add_raw_kv(to_json_iobuf(std::move(key)), std::nullopt);
+        } else {
+            auto key = schema_key{
+              .seq{s.seq}, .node{s.node}, .sub{sub}, .version{s.version}};
+            keys.push_back(key);
+            rb.add_raw_kv(to_json_iobuf(std::move(key)), std::nullopt);
+        }
+    }
 
-        auto key = schema_key{
-          .seq{write_at},
-          .node{my_node_id},
-          .sub{sub},
-          .version{projected.version}};
-        auto value = schema_value{
-          .sub{std::move(sub)},
-          .version{projected.version},
-          .type = type,
-          .id{projected.id},
-          .schema{std::move(def)},
-          .deleted = is_deleted::no};
+    // If a subject is in the store, it must have been replayed some somewhere,
+    // so there must be some entries in the list of keys to tombstone.
+    assert(!keys.empty());
 
-        auto batch = as_record_batch(key, value);
+    // Produce tombstones.  We do not need to check where they landed, because
+    // these can arrive in any order and be safely repeated.
+    auto batch = std::move(rb).build();
+    assert(batch.record_count() > 0);
 
-        kafka::partition_produce_response res
-          = co_await _client.local().produce_record_batch(
-            model::schema_registry_internal_tp, std::move(batch));
+    kafka::partition_produce_response res
+      = co_await _client.local().produce_record_batch(
+        model::schema_registry_internal_tp, std::move(batch));
+    if (res.error_code != kafka::error_code::none) {
+        vlog(
+          plog.error,
+          "Error writing to schema topic: {} {}",
+          res.error_code,
+          res.error_message);
+        throw kafka::exception(res.error_code, *res.error_message);
+    }
 
-        // TODO(Ben): Check the error reporting here
-        if (res.error_code != kafka::error_code::none) {
-            throw kafka::exception(res.error_code, *res.error_message);
+    // Replay the persisted deletions into our store
+    auto applier = consume_to_store(_store);
+    auto offset = res.base_offset;
+    for (auto k : keys) {
+        if (auto skey = std::get_if<schema_key>(&k)) {
+            co_await applier.apply(offset, *skey, std::nullopt);
+        } else if (auto dkey = std::get_if<delete_subject_key>(&k)) {
+            co_await applier.apply(offset, *dkey, std::nullopt);
         }
+        co_await _store.replay(offset);
+        offset++;
+    }
+
+    // TODO: deleting config_key entries, need to remember their sequence
+    // numbers too.  Actually... do config_key entries really need
+    // to be strictly ordered?  We're not allocating anything.
```

Comment:
Yes, config entries must be ordered, since they are last write wins, both per subject and globally.